Grafo.cpp

```diff
@@ -107,4 +107,170 @@ void Grafo::insertarArista(string origen, string destino, int precio, float tiem
     }
     else
         cout<<"\n\tYa existe una ruta que conecta " << origen << " con " << destino;
-}
\ No newline at end of file
+
+
+
+Matriz<string>* Grafo::generarRecorrido()
+{
+    Matriz<string>* recorridoMatriz = new Matriz<string>(NONE,elementos,elementos);//MATRIZ NxN CON TODOS NONE
+
+    for(int i = 0; i < elementos; i++)
+    {
+        for(int j = 0 ; j < elementos ; j++ )
+        {
+            if(i != j)
+            {
+                string aeropuerto = vertices->obtenerDato(j);
+                recorridoMatriz->modificarElemento(aeropuerto,i,j);
+            }
+        }
+
+    }
+
+    return recorridoMatriz;
+}
+
+
+Matriz<string>* Grafo::caminoMinimo(Matriz<int>* &precioMatriz)
+{
+    //Inicializo la matriz de Recorrido ...
+    Matriz<string>* recorridoMatriz = generarRecorrido();  ///LIBERAR DESPUES
+
+    //k = es el intermediario -> la letra de base
+    for(int k = 0 ; k < elementos ,k ++)
+    {
+        for(int i = 0 ;  i < elementos ; i ++)
+        {
+            for(int j = 0 ; j < elementos ; j++)
+            {
+                //Distancia acumulada
+                int distancia = precioMatriz->obtenerValor(i,k) + precioMatriz->obtenerValor(k,j);
+
+                if(distancia < precioMatriz->obtenerValor(i,j))
+                    precioMatriz->modificarElemento(distancia,i,j); //Modifico la el valor
+
+                    aeropuerto = vertices->obtenerDato(k);
+                    recorridoMatriz->modificarElemento(aeropuerto,i,j) //Modifico el aeropuerto
+
+            }
+        }
+    }
+    return recorridoMatriz;
+
+}
+
+
+Matriz<string>* Grafo::caminoMinimo(Matriz<float>* &tiempoMatriz)
+{
+    //Inicializo la matriz de Recorrido ...
+    Matriz<string>* recorridoMatriz = generarRecorrido();  ///LIBERAR DESPUES
+
+    //k = es el intermediario -> la letra de base
+    for(int k = 0 ; k < elementos ,k ++)
+    {
+        for(int i = 0 ;  i < elementos ; i ++)
+        {
+            for(int j = 0 ; j < elementos ; j++)
+            {
+                //Distancia acumulada
+                int distancia = tiempoMatriz->obtenerValor(i,k) + tiempoMatriz->obtenerValor(k,j);
+
+                if(distancia < tiempoMatriz->obtenerValor(i,j))
+                    tiempoMatriz->modificarElemento(distancia,i,j); //Modifico la el valor
+
+                    aeropuerto = vertices->obtenerDato(k);
+                    recorridoMatriz->modificarElemento(aeropuerto,i,j) //Modifico el aeropuerto
+
+            }
+        }
+    }
+
+    return recorridoMatriz;
+}
+
+Lista<string>* Grafo::armarEscalas(int posicionOrigen , int posDestino , Matriz<string>* &recorridoMatriz)
+{
+    Lista<string>* escalas = new Lista<string> ; //Lista que posee las escalas
+
+    string escala = recorridoMatriz->obtenerValor(posicionOrigen,posicionDestino);
```

Comment:
- Difieren los nombres de los parámetros recibidos y las variables utilizadas
- Creo que estaría bueno validar que tanto `posOrigen` como `posDestino` estén dentro del rango `0 <= pos < elementos`
```suggestion
Lista<string>* Grafo::armarEscalas(int posOrigen, int posDestino, Matriz<string>* &recorridoMatriz)
{
    Lista<string>* escalas = new Lista<string>;
    string escala = recorridoMatriz->obtenerValor(posOrigen, posDestino);
```

Reply:
Pensaba en validarlos en el menu , cuando el usuario ingresa los aeropuertos

Reply:
Que ponga aeropuertos que se encuentran en la lista de vertices

Reply:
Oki